keyboards/acheron/austin/keyboard.c

```diff
@@ -0,0 +1,297 @@
+#include "keyboard.h"
+#include "hal.h"
+#include "led_custom.h"
+#include "util.h"
+#include "quantum.h"
+
+//#include "ws2812.h"
+
+#include "raw_hid.h"
+#include "dynamic_keymap.h"
+#include "tmk_core/common/eeprom.h"
+
+backlight_config_t kb_backlight_config = {
+  .enable = true,
+  .breathing = true,
+  .level = BACKLIGHT_LEVELS
+};
+
+bool eeprom_is_valid(void)
+{
+	return (eeprom_read_word(((void*)EEPROM_MAGIC_ADDR)) == EEPROM_MAGIC &&
+			eeprom_read_byte(((void*)EEPROM_VERSION_ADDR)) == EEPROM_VERSION);
+}
+
+void eeprom_set_valid(bool valid)
+{
+	eeprom_update_word(((void*)EEPROM_MAGIC_ADDR), valid ? EEPROM_MAGIC : 0xFFFF);
+	eeprom_update_byte(((void*)EEPROM_VERSION_ADDR), valid ? EEPROM_VERSION : 0xFF);
+}
+
+void eeprom_reset(void)
+{
+	eeprom_set_valid(false);
+	eeconfig_disable();
+}
+
+void save_backlight_config_to_eeprom(){
+  eeprom_update_byte((uint8_t*)EEPROM_CUSTOM_BACKLIGHT, kb_backlight_config.raw);
+}
+
+void load_custom_config(){
+  kb_backlight_config.raw = eeprom_read_byte((uint8_t*)EEPROM_CUSTOM_BACKLIGHT);
+}
+
+#ifdef DYNAMIC_KEYMAP_ENABLE
+void dynamic_keymap_custom_reset(void){
+    void *p = (void*)(EEPROM_CUSTOM_BACKLIGHT);
+	void *end = (void*)(DYNAMIC_KEYMAP_MACRO_EEPROM_ADDR);
+	while ( p != end ) {
+		eeprom_update_byte(p, 0);
+		++p;
+	}
+}
+#endif
+
+void eeprom_init_kb(void)
+{
+	// If the EEPROM has the magic, the data is good.
+	// OK to load from EEPROM.
+	if (eeprom_is_valid()) {
+		load_custom_config();
+	} else	{
+#ifdef DYNAMIC_KEYMAP_ENABLE
+		// This resets the keymaps in EEPROM to what is in flash.
+		dynamic_keymap_reset();
+		// This resets the macros in EEPROM to nothing.
+		dynamic_keymap_macro_reset();
+        // Reset the custom stuff
+        dynamic_keymap_custom_reset();
+#endif
+		// Save the magic number last, in case saving was interrupted
+        save_backlight_config_to_eeprom();
+		eeprom_set_valid(true);
+	}
+}
```

Comment:
This needs refactoring, most of it is redundant and handled by `via.c`, the specific overrides for storing the custom backlight value can be integrated into how EEPROM gets used by VIA. See `keyboards/cannonkeys/stm32f072/keyboard.c` as an example.